src/Features/CSharp/Portable/NameTupleElement/CSharpNameTupleElementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.NameTupleElement;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.NameTupleElement
+{
+    [ExtensionOrder(After = PredefinedCodeRefactoringProviderNames.IntroduceVariable)]
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpNameTupleElementCodeRefactoringProvider)), Shared]
+    internal class CSharpNameTupleElementCodeRefactoringProvider : AbstractNameTupleElementCodeRefactoringProvider<ArgumentSyntax, TupleExpressionSyntax>
+    {
+        public CSharpNameTupleElementCodeRefactoringProvider()
+        {
+        }
+
+        protected override bool IsCloseParenOrComma(SyntaxToken token)
+            => token.IsKind(SyntaxKind.CloseParenToken, SyntaxKind.CommaToken);
```

Comment:
you need @Neme12's ISyntaxKindsService :)

Reply:
Maybe not.  Maybe if you can review his PR it can get in soon and then you can use it :)

Reply:
It's not (and I assume it won't be soon, it's quite a big PR and hard to review).